File: presentation/src/main/java/com/dagger/component/AppComponent.java
package com.dagger.component;

import com.dagger.module.ActivityModule;
import com.dagger.module.AppModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {

    ActivitySubComponent add(ActivityModule activityModule);

}

File: presentation/src/main/java/com/dagger/component/AppComponent.kt
package com.dagger.component

import com.dagger.module.ActivityModule
import com.dagger.module.AppModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {
    fun add(activityModule: ActivityModule): ActivitySubComponent
}